tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -167,13 +168,30 @@ static ClustersAndEntries MakeClusters(const std::vector<std::string> &treeNames
       std::vector<EntryCluster> clusters;
       while ((start = clusterIter()) < entries) {
          end = clusterIter.GetNextEntry();
-         // Add the current file's offset to start and end to make them (chain) global
-         clusters.emplace_back(EntryCluster{start + offset, end + offset});
+         // Currently, if a user specified a range, the clusters will be only globally obtained
+         // Hence, the guarantee that currentStart and currentEnd are always as expected
+         // That are: the cluster start must be at least startEntry, and at most the endEntry
+         // The current file's offset to start and end is added to make them (chain) global
+         const auto currentStart = std::max(start + offset, startEntry);
+         const auto currentEnd = std::min(end + offset, endEntry);
+         // This is not satified if the desired start is larger than the last entry of some cluster
+         // In this case, this cluster is not going to be processes further
+         if (currentStart < currentEnd)
+            clusters.emplace_back(EntryCluster{currentStart, currentEnd});
```

Comment:
I think it is really confusing to still call these "clusters". They are not really at cluster boundary anymore because we need to take into account the user-provided range of entries

Reply:
Hmm. Alternative is a `subTree` I think. But it might be the case that there is only 1 tree in a file, and is so long that is split into distinct clusers/subtrees, which might get split because of the range. So I have no opinion on the naming here.

Reply:
`EntryRange`?

Reply:
EntryRange?